src/main/java/core/basesyntax/report/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.report;
+
+import core.basesyntax.operation.ShopService;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private ShopService shopService;
+
+    @Override
+    public String generateReport() {
+        StringBuilder builder = new StringBuilder("Fruit, Quantity\n");
+        for (Map.Entry<String, Integer> entry: shopService.getFruits().entrySet()) {

```

Comment:
Use data structures from the Collection framework. Here, you are using a Map to represent the fruit storage. Think of the one that will be the most suitable for your needs.